### PR TITLE
Add documentation about taxons for end users

### PR DESCRIPTION
**Pull request by benjaminwil**
This adds documentation about taxons for end users. The audience for this documentation is store administrators or anyone who has to touch the `solidus_backend` as a business person.

### CODE REVIEW

File: guides/source/users/products/product-properties.html.md
Comment (gmacdougall) [inline guides/source/users/products/product-properties.html.md]: Typo here.

File: guides/source/users/products/taxonomies.html.md
Comment (gmacdougall) [inline guides/source/users/products/taxonomies.html.md]: The singular `"Brand"` doesn't match the plural `"Brands"` below.

**Review by gmacdougall — CHANGES_REQUESTED** [review]

File: guides/source/users/products/product-properties.html.md
Comment (benjaminwil) [reply guides/source/users/products/product-properties.html.md]: This file is no longer being changed on the branch.

I think I messed up a rebase in the past, which caused this change to show up. But I can't remember this far back.

**Comment by kennyadsl** [comment]
@benjaminwil Hey there, what's the status of this one? Do you think we can try to have it rebased and merged? If there's something to complete here and you don't have time, I can do that, just ask!

Thanks!

**Comment by benjaminwil** [comment]
@kennyadsl Should be good to go. It's been rebased, just needs a review from someone on the core team.

**Review by gmacdougall — APPROVED** [review]

**Review by kennyadsl — APPROVED** [review]
Thanks @benjaminwil !

**Pull request merged**
